lemur/plugins/lemur_acme/plugin.py

```diff
@@ -197,11 +245,38 @@ def create_certificate(self, csr, issuer_options):
         :param issuer_options:
         :return: :raise Exception:
         """
-        current_app.logger.debug("Requesting a new acme certificate: {0}".format(issuer_options))
-        acme_client, registration = setup_acme_client()
-        account_number = current_app.config.get('ACME_AWS_ACCOUNT_NUMBER')
+        authority = issuer_options.get('authority')
+        create_immediately = issuer_options.get('create_immediately', False)
+        acme_client, registration = setup_acme_client(authority)
+        dns_provider_d = issuer_options.get('dns_provider')
+        if not dns_provider_d:
+            raise Exception("DNS Provider setting is required for ACME certificates.")
+        dns_provider = dns_provider_service.get(dns_provider_d.get("id"))
+        credentials = json.loads(dns_provider.credentials)
+
+        current_app.logger.debug("Using DNS provider: {0}".format(dns_provider.provider_type))
+        dns_provider_type = __import__(dns_provider.provider_type, globals(), locals(), [], 1)
```

Comment:
What is this doing exactly?

Reply:
Are the use of locals() and globals() needed here or is merely convenience?

Reply:
Also, can/should this be done at start time vs. runtime?

Reply:
I changed the functionality of this to the below (diff pending). I can't preload without some refactoring due to flask.current_app
 
    def get_dns_provider(self, type):
        from lemur.plugins.lemur_acme import cloudflare, dyn, route53
        provider_types = {
            'cloudflare': cloudflare,
            'dyn': dyn,
            'route53': route53,
        }
        return provider_types[type]